numpy/core/src/umath/loops.c.src

```diff
@@ -1619,6 +1619,28 @@ TIMEDELTA_mm_m_remainder(char **args, npy_intp *dimensions, npy_intp *steps, voi
     }
 }
 
+NPY_NO_EXPORT void
+TIMEDELTA_mm_q_floor_divide(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
+{
+    void *NPY_UNUSED;
+    BINARY_LOOP {
+        const npy_timedelta in1 = *(npy_timedelta *)ip1;
+        const npy_timedelta in2 = *(npy_timedelta *)ip2;
+        if (in1 == NPY_DATETIME_NAT ||
+            in2 == NPY_DATETIME_NAT ||
+            in2 == 0) {
+            npy_set_floatstatus_divbyzero();
+            *((npy_int64 *)op1) = 0;
+        }
+        else {
+            LONGLONG_divide(args,
```

Comment:
I don't think this will work - you'll end up nesting the `BINARY_LOOP`s with this approach, doing an O(n) operation in O(n^2).

I think your best options for now is just to duplicate the contents of the long function.

If you want to go further, you could add `npy_floordiv`, `npy_floordivl` and `npy_floordivll` to npy_math, and then reuse those functions in the loops.

Reply:
I think that's why I had the cruder solution initially -- I couldn't find any other examples of loop nesting. The .src files designed to reduce excessive type-related code duplication are in some ways prone to code duplication themselves it seems.

Reply:
Removed the nested binary looping and replaced with code similar to longlong inner loop, though the div by 0 check can be safely removed since I was already checking that